hedera-mirror-monitor/src/test/java/com/hedera/mirror/monitor/publish/transaction/consensus/ConsensusSubmitMessageTransactionSupplierTest.java

```diff
@@ -0,0 +1,83 @@
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+import com.hedera.hashgraph.sdk.Hbar;
+import com.hedera.hashgraph.sdk.TopicMessageSubmitTransaction;
+import com.hedera.mirror.monitor.publish.transaction.AbstractTransactionSupplierTest;
+
+class ConsensusSubmitMessageTransactionSupplierTest extends AbstractTransactionSupplierTest {
+
+    private static final Hbar MAX_TRANSACTION_FEE_HBAR = Hbar.fromTinybars(1_000_000);
+
+    @Test
+    void createWithMinimumData() {
+        ConsensusSubmitMessageTransactionSupplier consensusSubmitMessageTransactionSupplier =
+                new ConsensusSubmitMessageTransactionSupplier();
+        consensusSubmitMessageTransactionSupplier.setTopicId(TOPIC_ID.toString());
+        TopicMessageSubmitTransaction actual = consensusSubmitMessageTransactionSupplier.get();
+
+        assertThat(actual)
+                .returns(MAX_TRANSACTION_FEE_HBAR, TopicMessageSubmitTransaction::getMaxTransactionFee)
+                .returns(TOPIC_ID, TopicMessageSubmitTransaction::getTopicId)
+                .satisfies(a -> assertThat(a.getMessage()).isNotNull())
+                .satisfies(a -> assertThat(a.getMessage().size()).isEqualTo(256));
+    }
+
+    @Test
+    void createWithCustomMessageSize() {
+        ConsensusSubmitMessageTransactionSupplier consensusSubmitMessageTransactionSupplier =
+                new ConsensusSubmitMessageTransactionSupplier();
+        consensusSubmitMessageTransactionSupplier.setMaxTransactionFee(1);
+        consensusSubmitMessageTransactionSupplier.setMessageSize(14);
+        consensusSubmitMessageTransactionSupplier.setTopicId(TOPIC_ID.toString());
+        TopicMessageSubmitTransaction actual = consensusSubmitMessageTransactionSupplier.get();
+
+        assertThat(actual)
+                .returns(ONE_TINYBAR, TopicMessageSubmitTransaction::getMaxTransactionFee)
+                .returns(TOPIC_ID, TopicMessageSubmitTransaction::getTopicId)
+                .satisfies(a -> assertThat(a.getMessage()).isNotNull())
+                .satisfies(a -> assertThat(a.getMessage().size()).isEqualTo(14));
+    }
+
+    @Test
+    void createWithCustomMessage() {
+        String message = "ConsensusSubmitMessageTransactionSupplierTest.createWithCustomData";
+        ConsensusSubmitMessageTransactionSupplier consensusSubmitMessageTransactionSupplier =
+                new ConsensusSubmitMessageTransactionSupplier();
+        consensusSubmitMessageTransactionSupplier.setMaxTransactionFee(1);
+        consensusSubmitMessageTransactionSupplier.setMessage(message);
+        consensusSubmitMessageTransactionSupplier.setTopicId(TOPIC_ID.toString());
+        TopicMessageSubmitTransaction actual = consensusSubmitMessageTransactionSupplier.get();
+
+        assertThat(actual)
+                .returns(true, a -> Arrays.equals(message.getBytes(StandardCharsets.UTF_8), actual.getMessage()
+                        .toByteArray()))
```

Comment:
Why wouldn't this work?
```suggestion
                .returns(ByteString.copyFromUtf8(message), TopicMessageSubmitTransaction::getMessage)
```

Reply:
Done.